x-pack/plugins/observability/public/components/shared/exploratory_view/utils/observability_index_patterns.ts

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { SavedObjectNotFound } from '../../../../../../../../src/plugins/kibana_utils/public';
+import {
+  DataPublicPluginStart,
+  IndexPattern,
+  FieldFormat as IFieldFormat,
+  IndexPatternSpec,
+} from '../../../../../../../../src/plugins/data/public';
+import { rumFieldFormats } from '../configurations/rum/field_formats';
+import { syntheticsFieldFormats } from '../configurations/synthetics/field_formats';
+import { FieldFormat, FieldFormatParams } from '../types';
+
+const appFieldFormats: Record<DataType, FieldFormat[] | null> = {
+  rum: rumFieldFormats,
+  apm: null,
+  logs: null,
+  metrics: null,
+  synthetics: syntheticsFieldFormats,
+};
+
+function getFieldFormatsForApp(app: DataType) {
+  return appFieldFormats[app];
+}
+
+export type DataType = 'synthetics' | 'apm' | 'logs' | 'metrics' | 'rum';
+
+export const indexPatternList: Record<DataType, string> = {
+  synthetics: 'synthetics_static_index_pattern_id',
+  apm: 'apm_static_index_pattern_id',
+  rum: 'rum_static_index_pattern_id',
+  logs: 'logs_static_index_pattern_id',
+  metrics: 'metrics_static_index_pattern_id',
+};
+
+const appToPatternMap: Record<DataType, string> = {
+  synthetics: '(synthetics-data-view)*,heartbeat-*,synthetics-*',
+  apm: 'apm-*',
+  rum: '(rum-data-view)*,apm-*',
+  logs: 'logs-*,filebeat-*',
+  metrics: 'metrics-*,metricbeat-*',
+};
+
+export function isParamsSame(param1: IFieldFormat['_params'], param2: FieldFormatParams) {
+  return (
+    param1?.inputFormat === param2?.inputFormat &&
+    param1?.outputFormat === param2?.outputFormat &&
+    param1?.showSuffix === param2?.showSuffix &&
+    param2?.outputPrecision === param1?.outputPrecision
+  );
+}
+
+export class ObservabilityIndexPatterns {
+  data?: DataPublicPluginStart;
+
+  constructor(data: DataPublicPluginStart) {
+    this.data = data;
+  }
+
+  async createIndexPattern(app: DataType) {
+    if (!this.data) {
+      throw new Error('data is not defined');
+    }
+
+    const pattern = appToPatternMap[app];
+
+    const fields = await this.data.indexPatterns.getFieldsForWildcard({
```

Comment:
Why are you looking up the fields like this? It shouldn't be necessary so I wonder if there's something I'm missing.

Reply:
i think you are right, i don't need to provide list of fields to create index pattern. i will emit this part, thank you taking a look

Reply:
Removed the code @mattkime